enumerables.rb

```diff
@@ -0,0 +1,129 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength
+
+# Enumerable Module
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = *self
+    x = 0
+    while x < arr.size
+      yield(arr[x])
+      x += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |element|
+      yield(element, i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each do |i|
+      arr << i if yield(i)
+    end
+    arr
+  end
+
+  def my_all?(arg = nil)
+    flag = true
+    my_each do |i|
+      if block_given?
+        return flag = false unless yield(i)
+      elsif arg.instance_of?(Regexp)
+        return flag = false unless arg.match(i)
+      elsif !block_given? && arg.nil?
+        return flag = false unless i
+      else
+        return flag = false unless i.is_a?(arg)
+      end
+    end
+    flag
+  end
+
+  def my_any?(arg = nil)
+    flag = false
+    my_each do |i|
+      if block_given?
+        return flag = true if yield(i)
+      elsif arg.instance_of?(Regexp)
+        return flag = true if arg.match(i)
+      elsif !block_given? && arg.nil?
+        return flag = true if i
+      elsif i.is_a?(arg)
+        return flag = true
+      end
+    end
+    flag
+  end
+
+  def my_none?(arg = nil)
+    flag = true
+    my_each do |i|
+      if block_given?
+        return flag = false if yield(i)
+      elsif arg.instance_of?(Regexp)
+        return flag = false if arg.match(i)
+      elsif !block_given? && arg.nil?
+        return flag = false if i
+      elsif i.is_a?(arg)
+        return flag = false
+      end
+    end
+    flag
+  end
```

Comment:
#my_none?

- [ ] When a pattern other than Regex or a Class is given, it should return false only if one of the collection matches the pattern.